### PR TITLE
metrics show: extend NOTEs for show

### LINKED ISSUE #2285: Metrics show xpath does not override dvcfile xpath
I do have a Dvcfile with metrics, and for some period of time (a few commits) I defined a specific xpath in that file; Now, when I run a different xpath on `dvc metrics show`, the old xpath (and hense, different metric) shows up for the commits where dvcfile specified it:

```
>>> dvc metric show -x pe20

working tree:
        data/rentals/nyc/metrics.json: [88.15]
bin_bedroom:
        data/rentals/nyc/metrics.json: [88.25]
first-iteration:
        data/rentals/nyc/metrics.json: [88.13]
nhds:
        data/rentals/nyc/metrics.json: [7.45]
roosevelt:
        data/rentals/nyc/metrics.json: [7.11]
``` 

obviosly pe20 was > 88 for all the commits, it's just showing median error for the `nhds` and `roosevelt`

### PR DESCRIPTION
**Pull request by pared**
* [x] Have you followed the guidelines in our
      [Contributing document](https://dvc.org/doc/user-guide/contributing)?

* [x] Does your PR affect documented changes or does it add new functionality
      that should be documented? If yes, have you created a PR for
      [dvc.org](https://github.com/iterative/dvc.org) documenting it or at
      least opened an issue for it? If so, please add a link to it.

-----
Fixes #2285

### CODE REVIEW

File: dvc/repo/metrics/show.py [hunk 1]
```diff
@@ -188,9 +188,15 @@ def _collect_metrics(repo, path, recursive, typ, xpath, branch):
...
 
-        # NOTE this case assumes that metric has been modified with
-        # `dvc metric modify -t {} -x {}`, so stage file should contain
-        # info what type metric is and what is its xpath
+        # NOTE this case assumes:
+        # - that metric has been modified with `dvc metric modify -t {} -x {
+        # }` and stage file should contain info what type metric is and what
+        # is its xpath
+        #
+        # or
```
Comment (shcheklein) [inline dvc/repo/metrics/show.py]: it's `and` I believe. type must not be provided via CLI _and_ there should be a metric definition in the DVC-file

if type is provided via CLI it overrides everything, including xpath.

Also, I would check when we calculate type based on the file extension. I hope it is happening only after this code, down the stack.

File: dvc/repo/metrics/show.py [hunk 2]
```diff
@@ -188,9 +188,15 @@ def _collect_metrics(repo, path, recursive, typ, xpath, branch):
         if not o.metric:
             continue
 
-        # NOTE this case assumes that metric has been modified with
-        # `dvc metric modify -t {} -x {}`, so stage file should contain
-        # info what type metric is and what is its xpath
+        # NOTE this case assumes:
+        # - that metric has been modified with `dvc metric modify -t {} -x {
```
Comment (shcheklein) [inline dvc/repo/metrics/show.py]: it's not that metric is modified, it's more like "there are type and path defined in the DVC-file for this metric". There are other way to modify the file, first of all. Second, I can modify it and remove the previous definition.

File: dvc/repo/metrics/show.py (hunk 1 above)
Comment (pared) [reply dvc/repo/metrics/show.py]: Yes, it is happening inside `_read_metrics` which is happening righ after `_collect_metrics`.

File: dvc/repo/metrics/show.py (hunk 2 above)
Comment (pared) [reply dvc/repo/metrics/show.py]: I removed also xpath, typ is core thing here. Xpath is explained later

File: dvc/repo/metrics/show.py [hunk 3]
```diff
@@ -188,9 +188,14 @@ def _collect_metrics(repo, path, recursive, typ, xpath, branch):
...
 
-        # NOTE this case assumes that metric has been modified with
-        # `dvc metric modify -t {} -x {}`, so stage file should contain
-        # info what type metric is and what is its xpath
+        # NOTE this case assumes:
+        # - that typ has not been provided in CLI call
+        #
+        # and one of following cases:
+        # - stage file contains info what type metric is
+        #
```
Comment (efiop) [inline dvc/repo/metrics/show.py]:
```suggestion
```

File: dvc/repo/metrics/show.py [hunk 4]
```diff
@@ -188,9 +188,14 @@ def _collect_metrics(repo, path, recursive, typ, xpath, branch):
...
             continue
 
-        # NOTE this case assumes that metric has been modified with
-        # `dvc metric modify -t {} -x {}`, so stage file should contain
-        # info what type metric is and what is its xpath
+        # NOTE this case assumes:
+        # - that typ has not been provided in CLI call
+        #
+        # and one of following cases:
+        # - stage file contains info what type metric is
```
Comment (efiop) [inline dvc/repo/metrics/show.py]:
```suggestion
        # - stage file contains metric type
```

File: dvc/repo/metrics/show.py [hunk 5]
```diff
@@ -188,9 +188,14 @@ def _collect_metrics(repo, path, recursive, typ, xpath, branch):
         if not o.metric:
             continue
 
-        # NOTE this case assumes that metric has been modified with
-        # `dvc metric modify -t {} -x {}`, so stage file should contain
-        # info what type metric is and what is its xpath
+        # NOTE this case assumes:
```
Comment (efiop) [inline dvc/repo/metrics/show.py]: Single-entry list is not needed :) 
```suggestion
        # NOTE this case assumes that typ has not been provided in CLI call and one of the following cases:
```

**Comment by shcheklein** [comment]
LGTM! 🎉 thanks!

**Pull request merged**